Templating/VOKTemplateModel.m

```diff
@@ -8,202 +8,184 @@
+    NSString *kitName;
+    BOOL isMac;
     if ([className isEqualToString:VOKTemplatingClassNameIOS]) {
-        self.kitName = KitNameIOS;
-        self.isMac = NO;
+        kitName = KitNameIOS;
+        isMac = NO;
     } else if ([className isEqualToString:VOKTemplatingClassNameMac]) {
-        self.kitName = KitNameMac;
-        self.isMac = YES;
+        kitName = KitNameMac;
+        isMac = YES;
+    } else {
+        return @"";
     }
-    NSString *result = [GRMustacheTemplate renderObject:self
-                                             fromString:templateString
-                                                  error:NULL];
-    self.imageClass = nil;
-    self.kitName = nil;
-    self.isMac = NO;
+    [template extendBaseContextWithObject:@{
+                                            @"frameworkPrefix": FrameworkPrefix,
+                                            @"constantStructName": ConstantStructName,
+                                            @"isMac": @(isMac),
+                                            @"kitName": kitName,
+                                            @"imageClass": className,
+                                            @"fileName": fileName,
+                                            }];
+    NSString *result = [template renderObject:self
+                                        error:NULL];
     return result;
 }
 
+- (GRMustacheTemplate *)templateWithFileName:(NSString *)fileName
+{
+    return [GRMustacheTemplate templateFromContentsOfURL:[self.templateDirectoryURL
+                                                          URLByAppendingPathComponent:fileName]
+                                                   error:NULL];
+}
+
 - (NSString *)renderObjCHWithClassName:(NSString *)className
+                              fileName:(NSString *)fileName
 {
-    return [self renderWithClassName:className templateString:MustacheFileH];
+    return [self renderWithClassName:className
+                            fileName:fileName
+                            template:[self templateWithFileName:@"ObjC.h.file.mustache"]];
 }
 
 - (NSString *)renderObjCMWithClassName:(NSString *)className
+                              fileName:(NSString *)fileName
 {
-    return [self renderWithClassName:className templateString:MustacheFileM];
+    return [self renderWithClassName:className
+                            fileName:fileName
+                            template:[self templateWithFileName:@"ObjC.m.file.mustache"]];
 }
 
 - (NSString *)renderSwiftWithClassName:(NSString *)className
+                              fileName:(NSString *)fileName
+{
+    return [self renderWithClassName:className
+                            fileName:fileName
+                            template:[self templateWithFileName:@"swift.file.mustache"]];
+}
+
+#pragma mark - Embedded Template Handling
+
+- (NSData *)getEmbeddedTemplateData
+{
+    unsigned long size;
+    void *ptr = getsectiondata(&_mh_execute_header, "__TEXT",
```

Comment:
Can you throw in a comment here for those of us not familiar with this Carbon-ite API so we know what's happening here?